src/main/scala/scala/swing/Publisher.scala

```diff
@@ -132,42 +134,52 @@ private[swing] class StrongReference[+T <: AnyRef](value: T) extends Reference[T
     override def toString: String = get.map(_.toString).getOrElse("<deleted>")
     def clear(): Unit = { ref = None }
     def enqueue(): Boolean = false
-    def isEnqueued(): Boolean = false
-  }
-
-abstract class RefBuffer[A <: AnyRef] extends mutable.Buffer[A] with SingleRefCollection[A] { self =>
-  protected val underlying: mutable.Buffer[Reference[A]]
-
-  def +=(el: A): this.type = { purgeReferences(); underlying += Ref(el); this }
-  def +=:(el: A): this.type = { purgeReferences(); Ref(el) +=: underlying; this }
-  def remove(el: A): Unit = { underlying -= Ref(el); purgeReferences(); }
-  def remove(n: Int): A = { val el = apply(n); remove(el); el }
-  def insertAll(n: Int, iter: Iterable[A]): Unit = {
-    purgeReferences()
-    underlying.insertAll(n, iter.view.map(Ref))
-  }
-  def update(n: Int, el: A): Unit = { purgeReferences(); underlying(n) = Ref(el) }
-  def apply(n: Int): A = {
-    purgeReferences()
-    var el = underlying(n).get
-    while (el.isEmpty) {
-      purgeReferences(); el = underlying(n).get
-    }
-    el.get
+    def isEnqueued: Boolean = false
   }
 
-  def length: Int = { purgeReferences(); underlying.length }
-  def clear(): Unit = { underlying.clear(); purgeReferences() }
```

Comment:
do we need to keep this commented section?

Reply:
Yes it should be removed. I had it commented to see if it will pop up in any use scenario, but so far I haven't seen any code using `RefBuffer`, so I guess it can be eliminated.

Reply:
Yes it should be removed. I had it commented to see if it will pop up in any use scenario, but so far I haven't seen any code using `RefBuffer`, so I guess it can be eliminated.